src/main/java/org/wildfly/naming/client/util/NamingUtils.java

```diff
@@ -57,7 +57,7 @@ public static CompositeName toCompositeName(final Name name) throws InvalidNameE
             return (CompositeName) name;
         } else {
             final CompositeName compositeName = new CompositeName();
-            compositeName.add(name.toString());
+            compositeName.addAll(new CompositeName(name.toString()));
```

Comment:
The reason for using CompositeName.toString() was originally to create a bidirectional mapping between name and String so we can avoid using serialization for simple cases.  I guess you're telling me that the mapping is not bidirectional though :)